include/dynamixel_control_hw/dynamixel_hardware_interface.hpp

```diff
@@ -212,47 +180,61 @@ namespace dynamixel {
+                    }
+
                     // tell ros_control the in-memory address to change to set new
-                    // position goal for the actuator
-                    hardware_interface::JointHandle pos_handle(
+                    // position or velocity goal for the actuator (depending on
+                    // hardware_mode)
+                    hardware_interface::JointHandle cmd_handle(
                         _jnt_state_interface.getHandle(dynamixel_iterator->second),
                         &_joint_commands[i]);
-                    _jnt_pos_interface.registerHandle(pos_handle);
-
-                    try {
-                        // enable the actuator
-                        dynamixel::StatusPacket<Protocol> status;
-                        ROS_DEBUG_STREAM("Enabling joint " << _servos[i]->id());
-                        _dynamixel_controller.send(_servos[i]->set_torque_enable(1));
-                        _dynamixel_controller.recv(status);
-
-                        // set max speed
-                        typename std::unordered_map<id_t, double>::iterator dynamixel_max_speed_iterator
-                            = _dynamixel_max_speed.find(_servos[i]->id());
-                        if (dynamixel_max_speed_iterator != _dynamixel_max_speed.end()) {
-                            dynamixel::StatusPacket<Protocol> status;
-                            ROS_DEBUG_STREAM("Setting velocity limit of joint "
-                                << _dynamixel_map[_servos[i]->id()] << " to "
-                                << dynamixel_max_speed_iterator->second << " rad/s.");
-                            _dynamixel_controller.send(_servos[i]->set_moving_speed_angle(dynamixel_max_speed_iterator->second));
-                            _dynamixel_controller.recv(status);
-                        }
+                    if (OperatingMode::joint == hardware_mode) {
+                        _jnt_pos_interface.registerHandle(cmd_handle);
                     }
-                    catch (dynamixel::errors::Error& e) {
-                        ROS_ERROR_STREAM("Caught a Dynamixel exception while "
-                            << "initializing:\n"
-                            << e.msg());
+                    else if (OperatingMode::wheel == hardware_mode) {
+                        _jnt_vel_interface.registerHandle(cmd_handle);
                     }
+                    else if (OperatingMode::unknown != hardware_mode)
+                        ROS_ERROR_STREAM("Servo " << id << " was not initialised "
+                                                  << "(operating mode "
+                                                  << mode2str(hardware_mode)
+                                                  << " is not supported)");
+
+                    // enable torque output on the servo and set its configuration
+                    // including max speed
+                    _enable_and_configure_servo(_servos[i], hardware_mode);
```

Comment:
If I am not mistaken, when the `hardware_mode` is `OperatingModel::unknown` we should not enable the servo but from what I see we still `set_torque_enable(1)` in this function (`_enable_and_configure_servo`) no matter the mode. Of course, I might be missing something..

Reply:
You're right. Fixed